catalog/transformer/catalog-transformer-tika-input/src/test/java/ddf/catalog/transformer/input/tika/TikaInputTransformerTest.java

```diff
@@ -144,49 +189,51 @@ public void testRegisterService() {
+    public void testExtractors() throws Exception {
         InputStream stream = Thread.currentThread()
                 .getContextClassLoader()
                 .getResourceAsStream("testPDF.pdf");
-        tikaInputTransformer.addContentMetadataExtractors(serviceRef);
+        tikaInputTransformer.addContentMetadataExtractors(serviceRefCme);
+        tikaInputTransformer.addMetadataExtractors(serviceRefMe);
         Metacard metacard = tikaInputTransformer.transform(stream);
-
+        verify(cme).process(anyString(), anyObject());
+        verify(me).process(anyString(), anyObject());
+        verify(cme).getMetacardAttributes();
+        verify(me).getMetacardAttributes();
         assertThat(metacard.getMetacardType()
                 .getName(), is(PDF_METACARDTYPE_NAME));
-        int matchedAttrs = (int) metacard.getMetacardType()
+        List<String> actualNames = metacard.getMetacardType()
                 .getAttributeDescriptors()
                 .stream()
-                .filter(attributeDescriptors::contains)
-                .count();
-
-        assertThat(matchedAttrs, is(attributeDescriptors.size()));
-        tikaInputTransformer.removeContentMetadataExtractor(serviceRef);
+                .map(AttributeDescriptor::getName)
+                .collect(toList());
+        assertThat("Missing an attribute descriptor",
+                actualNames,
+                allOf(hasItem(equalTo("attr1")),
+                        hasItem(equalTo("attr2")),
+                        hasItem(equalTo("attr3"))));
     }
 
     @Test
-    public void testContentExtractorRemoval() throws Exception {
+    public void testRemoveExtractors() throws Exception {
         InputStream stream = Thread.currentThread()
                 .getContextClassLoader()
                 .getResourceAsStream("testPDF.pdf");
-
-        tikaInputTransformer.addContentMetadataExtractors(serviceRef);
-        tikaInputTransformer.removeContentMetadataExtractor(serviceRef);
-        Metacard metacard = tikaInputTransformer.transform(stream);
-
-        assertThat(metacard.getMetacardType()
-                .getName(), is(PDF_METACARDTYPE_NAME));
-        int matchedAttrs = (int) metacard.getMetacardType()
-                .getAttributeDescriptors()
-                .stream()
-                .filter(attributeDescriptors::contains)
-                .count();
-
-        assertThat(matchedAttrs, is(0));
+        tikaInputTransformer.addContentMetadataExtractors(serviceRefCme);
+        tikaInputTransformer.addMetadataExtractors(serviceRefMe);
+        tikaInputTransformer.removeContentMetadataExtractor(serviceRefCme);
+        tikaInputTransformer.removeMetadataExtractor(serviceRefMe);
+        tikaInputTransformer.transform(stream);
+        verify(cme, never()).process(anyString(), anyObject());
+        verify(me, never()).process(anyString(), anyObject());
+        verify(cme, never()).getMetacardAttributes();
+        verify(me, never()).getMetacardAttributes();
```

Comment:
Included the new interface `MetadataExtractors` in the test.
Renamed the test
Changed assertion from checking the number of AttributeDescriptors on the MetacardType. Now uses Mockito's `verify` to make sure the  extractors were not invoked (and therefore must have been removed successfully).